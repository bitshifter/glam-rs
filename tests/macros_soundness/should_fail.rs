unsafe fn some_random_unsafe_fn() {}

macro_rules! test_soundness {
    ($m:ident ! ( $($arg:expr),* )) => {
        let _ = glam::$m!($({
            some_random_unsafe_fn();
            $arg
        }),*);
    }
}

fn main() {
    test_soundness!(const_m128!([1.0; 4]));
    test_soundness!(const_vec2!([1.0; 2]));
    test_soundness!(const_vec3!([1.0; 3]));
    test_soundness!(const_vec3a!([1.0; 3]));
    test_soundness!(const_vec4!([1.0; 4]));
    test_soundness!(const_mat2!([1.0; 2], [1.0; 2]));
    test_soundness!(const_mat2!([1.0; 4]));
    test_soundness!(const_mat3!([1.0; 3], [1.0; 3], [1.0; 3]));
    test_soundness!(const_mat3!([1.0; 9]));
    test_soundness!(const_mat3a!([1.0; 3], [1.0; 3], [1.0; 3]));
    test_soundness!(const_mat3a!([1.0; 9]));
    test_soundness!(const_mat4!([1.0; 4], [1.0; 4], [1.0; 4], [1.0; 4]));
    test_soundness!(const_mat4!([1.0; 16]));
    test_soundness!(const_quat!([1.0; 4]));
    test_soundness!(const_dvec2!([1.0; 2]));
    test_soundness!(const_dvec3!([1.0; 3]));
    test_soundness!(const_dvec4!([1.0; 4]));
    test_soundness!(const_dmat2!([1.0; 2], [1.0; 2]));
    test_soundness!(const_dmat2!([1.0; 4]));
    test_soundness!(const_dmat3!([1.0; 3], [1.0; 3], [1.0; 3]));
    test_soundness!(const_dmat3!([1.0; 9]));
    test_soundness!(const_dmat4!([1.0; 4], [1.0; 4], [1.0; 4], [1.0; 4]));
    test_soundness!(const_dmat4!([1.0; 16]));
    test_soundness!(const_dquat!([1.0; 4]));
    test_soundness!(const_ivec2!([1; 2]));
    test_soundness!(const_ivec3!([1; 3]));
    test_soundness!(const_ivec4!([1; 4]));
    test_soundness!(const_uvec2!([1; 2]));
    test_soundness!(const_uvec3!([1; 3]));
    test_soundness!(const_uvec4!([1; 4]));
}
