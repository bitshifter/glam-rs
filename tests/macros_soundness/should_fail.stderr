error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
13 |     test_soundness!(const_m128!([1.0; 4]));
   |     -------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
14 |     test_soundness!(const_vec2!([1.0; 2]));
   |     -------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
15 |     test_soundness!(const_vec3!([1.0; 3]));
   |     -------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
16 |     test_soundness!(const_vec3a!([1.0; 3]));
   |     --------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
17 |     test_soundness!(const_vec4!([1.0; 4]));
   |     -------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
18 |     test_soundness!(const_mat2!([1.0; 2], [1.0; 2]));
   |     ------------------------------------------------ in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
19 |     test_soundness!(const_mat2!([1.0; 4]));
   |     -------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
20 |     test_soundness!(const_mat3!([1.0; 3], [1.0; 3], [1.0; 3]));
   |     ---------------------------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
21 |     test_soundness!(const_mat3!([1.0; 9]));
   |     -------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
22 |     test_soundness!(const_mat3a!([1.0; 3], [1.0; 3], [1.0; 3]));
   |     ----------------------------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
23 |     test_soundness!(const_mat3a!([1.0; 9]));
   |     --------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
24 |     test_soundness!(const_mat4!([1.0; 4], [1.0; 4], [1.0; 4], [1.0; 4]));
   |     -------------------------------------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
25 |     test_soundness!(const_mat4!([1.0; 16]));
   |     --------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
26 |     test_soundness!(const_quat!([1.0; 4]));
   |     -------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
27 |     test_soundness!(const_dvec2!([1.0; 2]));
   |     --------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
28 |     test_soundness!(const_dvec3!([1.0; 3]));
   |     --------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
29 |     test_soundness!(const_dvec4!([1.0; 4]));
   |     --------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
30 |     test_soundness!(const_dmat2!([1.0; 2], [1.0; 2]));
   |     ------------------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
31 |     test_soundness!(const_dmat2!([1.0; 4]));
   |     --------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
32 |     test_soundness!(const_dmat3!([1.0; 3], [1.0; 3], [1.0; 3]));
   |     ----------------------------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
33 |     test_soundness!(const_dmat3!([1.0; 9]));
   |     --------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
34 |     test_soundness!(const_dmat4!([1.0; 4], [1.0; 4], [1.0; 4], [1.0; 4]));
   |     --------------------------------------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
35 |     test_soundness!(const_dmat4!([1.0; 16]));
   |     ---------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
36 |     test_soundness!(const_dquat!([1.0; 4]));
   |     --------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
37 |     test_soundness!(const_ivec2!([1; 2]));
   |     ------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
38 |     test_soundness!(const_ivec3!([1; 3]));
   |     ------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
39 |     test_soundness!(const_ivec4!([1; 4]));
   |     ------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
40 |     test_soundness!(const_uvec2!([1; 2]));
   |     ------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
41 |     test_soundness!(const_uvec3!([1; 3]));
   |     ------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> tests/macros_soundness/should_fail.rs:6:13
   |
6  |             some_random_unsafe_fn();
   |             ^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
42 |     test_soundness!(const_uvec4!([1; 4]));
   |     ------------------------------------- in this macro invocation
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
   = note: this error originates in the macro `test_soundness` (in Nightly builds, run with -Z macro-backtrace for more info)
